src/Game.cpp

```diff
@@ -348,12 +351,27 @@ void Game::Battle() {
         _Player->DisplayHUD(_Enemy);
         _Enemy->DisplayHUD();
         int damagePlayer = _Player->Action();
+        int extradamagePlayer = _Player->ReturnExtraMolotovDamage();
         // Player's turn to attack Enemy or choose other action.
-        if (damagePlayer != SKIP_TURN) {
+        if (damagePlayer != SKIP_TURN && damagePlayer != -3) {
             _Enemy->TakeDamage(damagePlayer);
             // Pauses console and ignores user input for SLEEP_MS milliseconds.
             Sleep(SLEEP_MS);
         }
+        // Player turn is over, enemy's status effect damage is calculated
+        if (0 < _Player->extradamageturns && _Player->extradamageturns <= 2 && damagePlayer != -1) {
+            _Player->extradamageturns--;
+            _Enemy->TakeDamage(extradamagePlayer);
+            _Player->ReturnDialog(extradamagePlayer, " damage points from burn dealt");
+        }
+        // Check if enemy has had molotov thrown at him before or now
+        if (damagePlayer == -3) {
+            damagePlayer = Common::RandomInt(30, 40);
+            _Enemy->TakeDamage(damagePlayer);
+            _Player->ReturnDialog(damagePlayer, " regular damage points dealt");
+            _Player->extradamageturns = 2;
+            _Player->ReturnDialog(_Player->extradamageturns, " turns of extra burn damage");
+        }
```

Comment:
There is still molotov logic in the game loop, we'd like to avoid that, I think this can be done wholly in the Player/Enemy classes. Let me know if you have any questions about that.

Reply:
I can reduce this to just one function call, if that is what you mean by "done wholly in the Player/Enemy classes". Otherwise, would you rather the molotov damage be added onto the main damage i.e. damagePlayer?

Reply:
Yup, the molotov damage should be added onto the main damage, then the dialog messages can be printed in the Player class' `UseMolotov`, similar to `UseBomb`, `BowAndArrow`, etc. Keeping track of `extradamageturns` should also be in the Player class.

Reply:
With that approach, you don't need to modify anything in the Enemy class. So really it should be done wholly in the Player class, if I'm not mistaken.